### PR TITLE
Fix: Add schema in Postgres Cache to swapping method

### LINKED ISSUE #148: 🐛 Bug: PostgreSQL Cache can't rename intermediate Table after replication
PyAirbyte Version: 0.8.1
Airbyte Source LinkedIn Pages Version: 1.0.2

I just tested replicating the LinkedIn Pages Source to a local PostgreSQL and the Connector failed. I tested with two different streams. `organization_lookup` and `follower_statistics`. Both times the Connector failed with the Error Message.

> ProgrammingError(psycopg2.errors.UndefinedTable) relation "<stream_name>" does not exist

Both of the Tables exist in the DB after running the Script but are empty. 

image

The number of runs doesn't matter. It happens when the table will be created first time but also when the table is already existing

**Sample Code**

```python
import airbyte as ab
from airbyte.caches import PostgresCache

cache = PostgresCache(
    host="localhost",
    username="postgres",
    password="pass123",
    database="postgres",
    port=5432
)

source: ab.Source = ab.get_source("source-linkedin-pages")

config = {
    "org_id": "redacted",
    "credentials": {
        "auth_method": "oAuth2.0",
        "client_id": "redacted",
        "client_secret": "redacted",
        "refresh_token": "redacted"
    }
}

source.set_config(config)
source.check()

source.select_streams("follower_statistics")
read_results = source.read(cache)
```

<details><summary>Error Log organization_lookup</summary>
<p>

```
Started `source-linkedin-pages` read operation at 11:11:38...
Failed `source-linkedin-pages` read operation at 11:11:48.
Traceback (most recent call last):
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/sources/base.py", line 740, in read
    cache.processor.process_airbyte_messages(
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/_processors/base.py", line 203, in process_airbyte_messages
    self.write_all_stream_data(
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/_processors/base.py", line 214, in write_all_stream_data
    self.write_stream_data(stream_name, write_strategy=write_strategy)
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/_processors/sql/base.py", line 542, in write_stream_data
    self._write_temp_table_to_final_table(
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/_processors/sql/base.py", line 695, in _write_temp_table_to_final_table
    self._swap_temp_table_with_final_table(
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/_processors/sql/base.py", line 800, in _swap_temp_table_with_final_table
    self._execute_sql(commands)
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/_processors/sql/base.py", line 608, in _execute_sql
    raise SQLRuntimeError(msg) from None  # from ex
airbyte._processors.sql.base.SQLRuntimeError: Error when executing SQL:
ALTER TABLE organization_lookup RENAME TO organization_lookup_deleteme;
ALTER TABLE organization_lookup_01ht28pnfnyp9hy9xzcj3f557x RENAME TO organization_lookup;
DROP TABLE organization_lookup_deleteme;
ProgrammingError(psycopg2.errors.UndefinedTable) relation "organization_lookup" does not exist

[SQL: ALTER TABLE organization_lookup RENAME TO organization_lookup_deleteme;
ALTER TABLE organization_lookup_01ht28pnfnyp9hy9xzcj3f557x RENAME TO organization_lookup;
DROP TABLE organization_lookup_deleteme;]
(Background on this error at: https://sqlalche.me/e/14/f405)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/sources/base.py", line 749, in read
    raise exc.AirbyteConnectorFailedError(
airbyte.exceptions.AirbyteConnectorFailedError: AirbyteConnectorFailedError: Connector failed.

Log output: 
        Starting syncing SourceLinkedinPages
    Marking stream organization_lookup as STARTED
    Syncing stream: organization_lookup 
    Marking stream organization_lookup as RUNNING
    Read 1 records from organization_lookup stream
    Marking stream organization_lookup as STOPPED
    Finished syncing organization_lookup
    SourceLinkedinPages runtimes:
    Syncing stream organization_lookup 0:00:07.137647
    Finished syncing SourceLinkedinPages
``` 

</p>
</details>

<details><summary>Error Log follower_stastics</summary>
<p>

```
Started `source-linkedin-pages` read operation at 11:17:02...
Failed `source-linkedin-pages` read operation at 11:17:11.
Traceback (most recent call last):
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/sources/base.py", line 740, in read
    cache.processor.process_airbyte_messages(
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/_processors/base.py", line 203, in process_airbyte_messages
    self.write_all_stream_data(
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/_processors/base.py", line 214, in write_all_stream_data
    self.write_stream_data(stream_name, write_strategy=write_strategy)
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/_processors/sql/base.py", line 542, in write_stream_data
    self._write_temp_table_to_final_table(
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/_processors/sql/base.py", line 695, in _write_temp_table_to_final_table
    self._swap_temp_table_with_final_table(
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/_processors/sql/base.py", line 800, in _swap_temp_table_with_final_table
    self._execute_sql(commands)
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/_processors/sql/base.py", line 608, in _execute_sql
    raise SQLRuntimeError(msg) from None  # from ex
airbyte._processors.sql.base.SQLRuntimeError: Error when executing SQL:
ALTER TABLE follower_statistics RENAME TO follower_statistics_deleteme;
ALTER TABLE follower_statistics_01ht290hb63tkzmhjws1e2xyc7 RENAME TO follower_statistics;
DROP TABLE follower_statistics_deleteme;
ProgrammingError(psycopg2.errors.UndefinedTable) relation "follower_statistics" does not exist

[SQL: ALTER TABLE follower_statistics RENAME TO follower_statistics_deleteme;
ALTER TABLE follower_statistics_01ht290hb63tkzmhjws1e2xyc7 RENAME TO follower_statistics;
DROP TABLE follower_statistics_deleteme;]
(Background on this error at: https://sqlalche.me/e/14/f405)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/tino/repos/PyAirbyte_tests/.venv/lib/python3.10/site-packages/airbyte/sources/base.py", line 749, in read
    raise exc.AirbyteConnectorFailedError(
airbyte.exceptions.AirbyteConnectorFailedError: AirbyteConnectorFailedError: Connector failed.

Log output: 
        Starting syncing SourceLinkedinPages
    Marking stream follower_statistics as STARTED
    Syncing stream: follower_statistics 
    Marking stream follower_statistics as RUNNING
    Read 1 records from follower_statistics stream
    Marking stream follower_statistics as STOPPED
    Finished syncing follower_statistics
    SourceLinkedinPages runtimes:
    Syncing stream follower_statistics 0:00:07.102127
    Finished syncing SourceLinkedinPages
```

</p>
</details>

### PR DESCRIPTION
**Pull request by tinomerl**
Fixes #148 

I dug a bit around and found out, that the Problem was, that the schema name was missing in the final swaping. I just added the method used at other places to add the schema.

I separated the statements in several lines so that i won't go over the defined Line Limit from 100.

Tests were all successful except for the integration Tests of some caches, since i don't have access to them.
image

Also tested the fix manuallay with the pokeapi source and the linkedinpages source mentioned in #148. Everything worked and data was in the final tables.

### CODE REVIEW

File: airbyte/_processors/sql/base.py [hunk 1]
```diff
@@ -794,9 +794,11 @@ def _swap_temp_table_with_final_table(
...
         commands = "\n".join(
             [
-                f"ALTER TABLE {final_table_name} RENAME TO {deletion_name};",
-                f"ALTER TABLE {temp_table_name} RENAME TO {final_table_name};",
-                f"DROP TABLE {deletion_name};",
+                f"ALTER TABLE {self._fully_qualified(final_table_name)} RENAME"
+                f"TO {deletion_name};",
+                f"ALTER TABLE {self._fully_qualified(temp_table_name)} RENAME"
+                f"TO {final_table_name};",
+                f"DROP TABLE {self._fully_qualified(deletion_name)};",
```
Comment (aaronsteers) [inline airbyte/_processors/sql/base.py]: Minor typo here. In meeting line length limits, the delimiting space was inadvertently truncated.

```suggestion
                f"ALTER TABLE {self._fully_qualified(final_table_name)} RENAME "
                f"TO {deletion_name};",
                f"ALTER TABLE {self._fully_qualified(temp_table_name)} RENAME "
                f"TO {final_table_name};",
                f"DROP TABLE {self._fully_qualified(deletion_name)};",
```

I will go ahead and auto-apply.

**Comment by tinomerl** [comment]
@aaronsteers - thanks! I'm happy to help. 

Oh darn. I saw the typo. Should have caught that one myself by looking at the very first runs. 🙈

The link for the rerun works for me and I can see the results.

Great. That means it's easier and faster for you guys to check PRs from forks right?

**Comment by aaronsteers** [comment]
@tinomerl - Just for my info - can you confirm if you can see results only, or also the detailed logs?

**Comment by aaronsteers** [comment]
> Great. That means it's easier and faster for you guys to check PRs from forks right?

Exactly that. 😄

**Comment by aaronsteers** [comment]
Will merge shortly.

Tests are green here in the PR (auto-skipping tests that require creds), and tests are also green on the manual workflow [here](https://github.com/airbytehq/PyAirbyte/actions/runs/8470606404) (which uses our creds to do the full suite of tests).

image

**Comment by tinomerl** [comment]
> @tinomerl - Just for my info - can you confirm if you can see results only, or also the detailed logs?

I can also see the logs of every step when clicking it. But only after it's finished, if this makes any difference.

**Pull request merged**